src/System.Memory/ref/System.Memory.cs

```diff
@@ -75,6 +75,8 @@ public static partial class MemoryExtensions
         public static System.ReadOnlySpan<char> TrimStart(this System.ReadOnlySpan<char> span, char trimChar) { throw null; }
         public static System.ReadOnlySpan<char> TrimStart(this System.ReadOnlySpan<char> span, System.ReadOnlySpan<char> trimChars) { throw null; }
         public static bool TryGetString(this System.ReadOnlyMemory<char> readOnlyMemory, out string text, out int start, out int length) { throw null; }
+        public static bool TryCast<TFrom, TTo>(this ReadOnlySpan<TFrom> source, out ReadOnlySpan<TTo> output) where TFrom : struct where TTo : struct { throw null; }
```

Comment:
As we talked in the other thread, it would be better to make these instance methods. I am not sure if you'd like to merge this and then separately DCR, or make the API change now.